docs/fsharp-cheatsheet.md

```diff
@@ -1,523 +1,630 @@
+# Functions
 
-Use the `let` keyword to define named functions.
+## `let` bindings
 
-    let add n1 n2 = n1 + n2
-    let subtract n1 n2 = n1 - n2
-    let negate num = -1 * num
-    let print num = printfn $"The number is: {num}"
+Use the `let` keyword to define named functions.
 
-### Pipe and Composition Operators
+```fsharp
+let add n1 n2 = n1 + n2
+let subtract n1 n2 = n1 - n2
+let negate num = -1 * num
+let print num = printfn $"The number is: {num}"
+```
+## Pipe and Composition Operators
 
 Pipe operator `|>` is used to chain functions and arguments together.
 
-    let addTwoSubtractTwoNegateAndPrint num =
-        num |> add 2 |> subtract 2 |> negate |> print
+```fsharp
+let addTwoSubtractTwoNegateAndPrint num =
+    num |> add 2 |> subtract 2 |> negate |> print
+```
 
 Composition operator `>>` is used to compose functions:
 
-    let addTwoSubtractTwoNegateAndPrint' =
-        add 2 >> subtract 2 >> negate >> print
+```fsharp
+let addTwoSubtractTwoNegateAndPrint' =
+    add 2 >> subtract 2 >> negate >> print
+```
 
 Caution: The output is the _last_ argument to the next function.
 
-    // `addTwoSubtractTwoNegateAndPrint 10` becomes:
-    10
-    |> add 2       //  2 + 10  = 12
-    |> subtract 2  //  2 - 12  = -10
-    |> negate      // -1 * -10 = 10
-    |> print       // "The number is 10"
+```fsharp
+// `addTwoSubtractTwoNegateAndPrint 10` becomes:
+10
+|> add 2       //  2 + 10  = 12
+|> subtract 2  //  2 - 12  = -10
+|> negate      // -1 * -10 = 10
+|> print       // "The number is 10"
+```
+
+## Anonymous functions
+
+Anonymous, or "lambda" functions, are denoted by the `fun` keyword and the arrow operator `->`.
+
+```fsharp
+let isDescending xs =
+    xs
+    |> List.pairwise
+    |> List.forAll (fun (x, y) -> x > y)
+
+let suspiciousRecords =
+    records
+    |> Seq.filter (fun x -> x.Age >= 150)
+```
+### _.Property shorthand
+
+If the lambda function has a single argument which is used in an atomic expression, the following shorthand has been available since F# 8:
+
+```fsharp
+let names =
+    people
+    |> List.map (fun person -> person.Name)   // regular lambda expression
+
+let names' =
+    people
+    |> List.map _.Name    // _.Property shorthand
+```
+
+You may chain properties and methods together, so long as there is no "space" in the expression. E.g.:
+
+```fsharp
+let uppercaseNames =
+    people |> List.map _.Name.ToUpper()
+```
 
 <div id="functions-unit-type"></div>
 
-### `unit` Type
+## `unit` Type
 
 The `unit` type is a type that indicates the absence of a specific value. It is represented by `()`.
 The most common use is when you have a function that receives no parameters, but you need it to evaluate on every call:
 
-    let appendSomeTextToFile () =  // without unit, only one line would be appended to the file
-        System.IO.File.AppendAllText($"{__SOURCE_DIRECTORY__}/file.txt", "New line")
+```fsharp
+let appendSomeTextToFile () =  // without unit, only one line would be appended to the file
```

Comment:
maybe a function that calls DateTime.Now is a better example? this is pety cryptic for a newcomer